specification/billing/resource-manager/Microsoft.Billing/stable/2023-04-01/transaction.json

```diff
@@ -63,259 +562,377 @@
         "billingProfileDisplayName": {
-          "description": "The name of the billing profile which will be billed for the transaction.",
-          "readOnly": true,
-          "type": "string"
+          "description": "The name of the billing profile."
         },
         "billingProfileId": {
-          "description": "The ID of the billing profile which will be billed for the transaction.",
-          "readOnly": true,
+          "description": "The fully qualified ID that uniquely identifies a billing profile.",
           "type": "string"
         },
+        "consumptionCommitmentDecremented": {
+          "description": "The amount of Microsoft Azure Consumption Commitment(MACC) decrement through the transaction.",
+          "allOf": [
+            {
+              "$ref": "./types.json#/definitions/Amount"
+            }
+          ]
+        },
         "customerDisplayName": {
-          "description": "The name of the customer for which the transaction took place. The field is applicable only for Microsoft Partner Agreement billing account.",
-          "readOnly": true,
+          "description": "The name of the customer.",
           "type": "string"
         },
         "customerId": {
-          "description": "The ID of the customer for which the transaction took place. The field is applicable only for Microsoft Partner Agreement billing account.",
-          "readOnly": true,
+          "description": "The fully qualified ID that uniquely identifies a customer.",
           "type": "string"
         },
+        "creditType": {
+          "description": "The credit type of the transaction. Applies only to credited transactions.",
+          "enum": [
+            "Other",
+            "AzureFreeCredit",
+            "AzureCreditOffer",
+            "ServiceInterruption",
+            "Refund"
+          ],
+          "type": "string",
+          "x-ms-enum": {
+            "name": "CreditType",
+            "modelAsString": true
+          }
+        },
         "date": {
           "description": "The date of transaction.",
-          "format": "date-time",
-          "readOnly": true,
           "type": "string"
         },
         "discount": {
+          "format": "double",
           "description": "The percentage discount, if any, applied to this transaction.",
-          "readOnly": true,
           "type": "number"
         },
         "effectivePrice": {
-          "$ref": "./types.json#/definitions/Amount",
           "description": "The price of the product after applying any discounts.",
-          "readOnly": true
+          "allOf": [
+            {
+              "$ref": "./types.json#/definitions/Amount"
+            }
+          ]
         },
         "exchangeRate": {
+          "format": "double",
           "description": "The exchange rate used to convert charged amount to billing currency, if applicable.",
-          "readOnly": true,
           "type": "number"
         },
         "invoice": {
```

Comment:
Investigate behavior of this field for billed and unbilled. Can we improve the description?

Reply:
This is the "InvoiceName" whereas "InvoiceId" is the fully qualified Arm ID. But we can't change it here.